src/Nethermind/Nethermind.Trie/VerkleTree.cs

```diff
@@ -109,56 +112,77 @@ public void SetValue(Span<byte> rawKey, byte[] value)
         // TODO; error handling here? or at least a way to check if the operation was successful
         RustVerkleLib.VerkleTrieInsert(_verkleTrieObj, rawKey.ToArray(), value);
     }
+    
+    [DebuggerStepThrough]
+    public void SetValue(Span<byte> keyPrefix, byte subIndex, byte[] value)
+    {
+        if (_logger.IsTrace)
+            _logger.Trace($"{(value.Length == 0 ? $"Deleting {keyPrefix.ToHexString() + subIndex}" : $"Setting {keyPrefix.ToHexString() + + subIndex} = {value.ToHexString()}")}");
+        // TODO; error handling here? or at least a way to check if the operation was successful
+        keyPrefix[31] = subIndex;
+        RustVerkleLib.VerkleTrieInsert(_verkleTrieObj, keyPrefix.ToArray(), value);
+    }
 
-    private byte[] GetTreeKeyPrefix(Address address, UInt256 treeIndex)
+    public byte[] GetTreeKeyPrefix(Address address, UInt256 treeIndex)
     {
-         // is it guaranteed that the its a 12 length byte array initialized with zeros?
-         byte[] addressPadding = new byte[12] ;
-         IEnumerable<byte> treeKeyPrecursor = addressPadding.Concat(address.Bytes);
-         treeKeyPrecursor = treeKeyPrecursor.Concat(treeIndex.ToBigEndian());
-         return Sha2.Compute(treeKeyPrecursor.ToArray());
+         byte[] keyPrefix = new byte[64];
+         Array.Copy(address.Bytes, 0, keyPrefix, 12, 20);
+         treeIndex.ToBigEndian(keyPrefix.AsSpan(32));
+         return Sha2.Compute(keyPrefix);
```

Comment:
If we could rewrite Sha2.Compute to take a Span<byte> and we probably can, we could stackallok keyPrefix, making it faster and eliminating GC.